File: CppPractice/CppLeetcode/Leetcode67.cpp
class Solution {
public:
    string addBinary(string a, string b) 
    {
        string ans;
        int carry = 0;
        reverse(a.begin(),a.end());
        reverse(b.begin(),b.end());
        int curA=0;
        int curB=0;
        while(curA<a.size()||curB<b.size()||carry)
        {
            int digit1=0;
            int digit2=0;
            if(curA<a.size())
            {
                if(a[curA]=='1'){digit1=1;}else{digit1=0;}
                curA++;
            }
            if(curB<b.size())
            {
                if(b[curB]=='1'){digit2=1;}else{digit2=0;}
                curB++;
            }
            ans+=to_string((digit1+digit2+carry)%2);
            carry = (digit1+digit2+carry)/2;
        }
        reverse(ans.begin(),ans.end());
        return ans;
        
    }
};

File: CSharpPractice/LeetcodePractice/Leetcode67.cs
public class Solution67 {
    public string ReverseString(string s)
    {
        char[] temp = s.ToCharArray();
        Array.Reverse(temp);
        return new string(temp);

    }
    public string AddBinary(string a, string b) 
    {
        a = ReverseString(a);
        b = ReverseString(b);
        List<string> ans=new List<string>();
        // Console.WriteLine(a);
        // Console.WriteLine(b);

        int carry =0;
        int aDigit=0;
        int bDigit=0;
        for(int i=0;i<Math.Max(a.Length,b.Length);i++)
        {
            if(i<a.Length && i<b.Length)
            {
                aDigit = a[i] - '0';
                bDigit = b[i] - '0';
                ans.Add(((aDigit + bDigit + carry)%2).ToString());
                carry = (aDigit + bDigit + carry)/2;
            }
            else if(i<a.Length)
            {
                aDigit = a[i] - '0';
                ans.Add(((aDigit + carry)%2).ToString());
                carry = (aDigit + carry)/2;

            }
            else if(i<b.Length)
            {
                bDigit = b[i] - '0';
                ans.Add((( bDigit + carry)%2).ToString());
                carry = (bDigit + carry)/2;
            }

        }
        if(carry>0){ans.Add("1");}
        // foreach(string e in ans)
        // {
        //     Console.WriteLine(e);
        // }
        ans.Reverse();
        ans.ToArray();
        return string.Join("", ans);
    }
}